source/client-side-operations-timeout/tests/global-timeoutMS.yml

```diff
@@ -0,0 +1,3142 @@
+              commandName: listDatabases
+              databaseName: admin
+              command:
+                listDatabases: 1
+                maxTimeMS: { $$exists: false }
+  - description: "timeoutMS can be configured on a MongoClient - createChangeStream on client"
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                uriOptions:
+                  timeoutMS: 50
+                useMultipleMongoses: false
+                observeEvents:
+                  - commandStartedEvent
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: ["aggregate"]
+              blockConnection: true
+              blockTimeMS: 60
+      - name: createChangeStream
+        object: *client
+        arguments:
+          pipeline: []
+          
+        expectError:
+          isTimeoutError: true
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: aggregate
+              databaseName: admin
+              command:
+                aggregate: 1
+                maxTimeMS: { $$type: ["int", "long"] }
+  - description: "timeoutMS can be set to 0 on a MongoClient - createChangeStream on client"
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                uriOptions:
+                  timeoutMS: 0
+                useMultipleMongoses: false
+                observeEvents:
+                  - commandStartedEvent
+                ignoreCommandMonitoringEvents:
+                  - killCursors
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: ["aggregate"]
+              blockConnection: true
+              blockTimeMS: 15
+      - name: createChangeStream
+        object: *client
+        arguments:
+          pipeline: []
```

Comment:
The Go driver expects a `saveResultAsEntity` field for all the `createChangeStream` operations in these new tests and errors in its absence. The unified test format does not seem to imply that there _must_ be a `saveResultAsEntity` field for `createChangeStream`, so I _could_ just throw away the result of the `watch` command, but wanted to point out that this deviates from existing `createChangeStream` operations (by nature of the template used to create the tests).

Reply:
Good catch. The unified test format says (emphasis mine):
> ... before the resulting change stream *_might be saved_* with [operation.saveResultAsEntity](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#operation-saveresultasentity).

So I think saveResultAsEntity can be optional. I tried adding saveResultAsEntity but it's tedious and error prone because it's mutually exclusive with expectError. Can we leave it optional for now?

Reply:
Not my intention to create additional work for anyone, but there may be value in introduce a new `valid-pass` test in the unified spec to demonstrate `createChangeStream` being used w/o `saveResultAsEntity` so that the only example of that usage isn't confined to the CSOT spec.

Feel free to ignore if this seems superfluous.

Reply:
Done.